confmap/provider.go

```diff
@@ -101,17 +102,39 @@ func WithRetrievedClose(closeFunc CloseFunc) RetrievedOption {
 }
 
 // NewRetrieved returns a new Retrieved instance that contains the data from the raw deserialized config.
-func NewRetrieved(rawConf map[string]interface{}, opts ...RetrievedOption) (Retrieved, error) {
+// The rawConf can be one of the following types:
+//   * Primitives: int, int32, int64, float32, float64, bool, string;
+//   * []interface{};
+//   * map[string]interface{};
+func NewRetrieved(rawConf interface{}, opts ...RetrievedOption) (Retrieved, error) {
+	if err := checkRawConfType(rawConf); err != nil {
+		return Retrieved{}, err
+	}
 	set := retrievedSettings{}
 	for _, opt := range opts {
 		opt(&set)
 	}
-	return Retrieved{conf: NewFromStringMap(rawConf), closeFunc: set.closeFunc}, nil
+	return Retrieved{rawConf: rawConf, closeFunc: set.closeFunc}, nil
 }
 
 // AsConf returns the retrieved configuration parsed as a Conf.
-func (r Retrieved) AsConf() (*Conf, error) {
-	return r.conf, nil
+func (r *Retrieved) AsConf() (*Conf, error) {
+	if r.rawConf == nil {
+		return New(), nil
+	}
+	val, ok := r.rawConf.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("retrieved value (type=%T) cannot be used as a Conf", r.rawConf)
+	}
+	return NewFromStringMap(val), nil
+}
+
+// AsRaw returns the retrieved configuration parsed as an interface{} which can be one of the following types:
+//   * Primitive float/int/bool/string;
```

Comment:
Line 106 says `Primitives: int, int32, int64, float32, float64, bool, string;`. Is the difference intentional?

Reply:
Mistake, fixed.